gridcapa-task-manager-app/src/main/java/com/farao_community/farao/gridcapa/task_manager/app/TaskManager.java

```diff
@@ -68,14 +69,34 @@ public Consumer<Flux<TaskStatusUpdate>> consumeTaskStatusUpdate() {
     }
 
     public void handleTaskStatusUpdate(TaskStatusUpdate taskStatusUpdate) {
-        Optional<Task> optionalTask = taskRepository.findByIdWithProcessFiles(taskStatusUpdate.getId());
-        if (optionalTask.isPresent()) {
-            Task task = optionalTask.get();
-            task.setStatus(taskStatusUpdate.getTaskStatus());
-            taskRepository.save(task);
-            taskUpdateNotifier.notify(task);
-        } else {
-            LOGGER.warn("Task {} does not exist. Impossible to update status", taskStatusUpdate.getId());
+        synchronized (LOCK) {
+            Optional<Task> optionalTask = taskRepository.findByIdWithProcessFiles(taskStatusUpdate.getId());
+            if (optionalTask.isPresent()) {
+                Task task = optionalTask.get();
+                task.setStatus(taskStatusUpdate.getTaskStatus());
+                taskRepository.saveAndFlush(task);
+                taskUpdateNotifier.notify(task);
+                LOGGER.debug("Task status has been updated on {} to {}", task.getTimestamp(), taskStatusUpdate.getTaskStatus());
+            } else {
+                LOGGER.warn("Task {} does not exist. Impossible to update status", taskStatusUpdate.getId());
+            }
+        }
+    }
+
+    public Optional<Task> handleTaskStatusUpdate(OffsetDateTime timestamp, TaskStatus taskStatus) {
+        synchronized (LOCK) {
```

Comment:
Perhaps you could mutualize code for the inner part of both handleTaskStatusUpdate methods